src/index/autotune.rs

```diff
@@ -0,0 +1,91 @@
+use std::ffi;
+use std::ptr;
+
+use super::*;
+
+use crate::error::Result;
+use crate::faiss_try;
+
+/// Uses a-priori knowledge on the Faiss indexes to extract tunable parameters.
+pub struct ParameterSpace {
+    inner: *mut FaissParameterSpace,
+}
+
+impl ParameterSpace {
+    /// Create a new instance
+    pub fn new() -> Result<Self> {
+        unsafe {
+            let mut inner = ptr::null_mut();
+            faiss_try(faiss_ParameterSpace_new(&mut inner))?;
+
+            Ok(ParameterSpace { inner })
+        }
+    }
+
+    /// Set one of the parameters
+    pub fn set_index_parameter(
+        &self,
+        index: &mut IndexImpl,
+        name: &ffi::CStr,
```

Comment:
Since this is a high level API, it is not idiomatic for one of the arguments to be a C-style string. Please make it a standard Rust string slice `&str` (or something that trivially converts to a string slice), then convert it internally, in a similar fashion to the `index_factory` function (https://github.com/Enet4/faiss-rs/blob/master/src/index/mod.rs#L374).